File: javascript/TennisGame2.js
export default class TennisGame2 {
  constructor(player1Name, player2Name) {
    this.P1point = 0;
    this.P2point = 0;
    this.P1res = "";
    this.P2res = "";
    this.player1Name = player1Name;
    this.player2Name = player2Name;
  }

  getScore() {
    var score = "";
    if (this.P1point === this.P2point && this.P1point < 3) {
      if (this.P1point === 0) score = "Love";
      if (this.P1point === 1) score = "Fifteen";
      if (this.P1point === 2) score = "Thirty";
      score += "-All";
    }
    if (this.P1point === this.P2point && this.P1point > 2) score = "Deuce";
    if (this.P1point > 0 && this.P2point === 0) {
      if (this.P1point === 1) this.P1res = "Fifteen";
      if (this.P1point === 2) this.P1res = "Thirty";
      if (this.P1point === 3) this.P1res = "Forty";
      this.P2res = "Love";
      score = this.P1res + "-" + this.P2res;
    }
    if (this.P2point > 0 && this.P1point === 0) {
      if (this.P2point === 1) this.P2res = "Fifteen";
      if (this.P2point === 2) this.P2res = "Thirty";
      if (this.P2point === 3) this.P2res = "Forty";
      this.P1res = "Love";
      score = this.P1res + "-" + this.P2res;
    }
    if (this.P1point > this.P2point && this.P1point < 4) {
      if (this.P1point === 2) this.P1res = "Thirty";
      if (this.P1point === 3) this.P1res = "Forty";
      if (this.P2point === 1) this.P2res = "Fifteen";
      if (this.P2point === 2) this.P2res = "Thirty";
      score = this.P1res + "-" + this.P2res;
    }
    if (this.P2point > this.P1point && this.P2point < 4) {
      if (this.P2point === 2) this.P2res = "Thirty";
      if (this.P2point === 3) this.P2res = "Forty";
      if (this.P1point === 1) this.P1res = "Fifteen";
      if (this.P1point === 2) this.P1res = "Thirty";
      score = this.P1res + "-" + this.P2res;
    }
    if (this.P1point > this.P2point && this.P2point >= 3) {
      score = "Advantage player1";
    }
    if (this.P2point > this.P1point && this.P1point >= 3) {
      score = "Advantage player2";
    }
    if (
      this.P1point >= 4 &&
      this.P2point >= 0 &&
      this.P1point - this.P2point >= 2
    ) {
      score = "Win for player1";
    }
    if (
      this.P2point >= 4 &&
      this.P1point >= 0 &&
      this.P2point - this.P1point >= 2
    ) {
      score = "Win for player2";
    }
    return score;
  }

  SetP1Score(number) {
    var i;
    for (i = 0; i < number; i++) {
      this.P1Score();
    }
  }

  SetP2Score(number) {
    var i;
    for (i = 0; i < number; i++) {
      this.P2Score();
    }
  }

  P1Score() {
    this.P1point++;
  }

  P2Score() {
    this.P2point++;
  }

  wonPoint(player) {
    if (player === "player1") this.P1Score();
    else this.P2Score();
  }
}

File: dart/lib/TennisGame2.dart

import 'dart:core';

import 'TennisGame.dart';

class TennisGame2 implements TennisGame
{
    int P1point = 0;
    int P2point = 0;

    String P1res = "";
    String P2res = "";
    String _player1Name;
    String _player2Name;

  TennisGame2(String this._player1Name, String this._player2Name);

    String getScore(){
        String score = "";
        if (P1point == P2point && P1point < 4)
        {
            if (P1point==0)
                score = "Love";
            if (P1point==1)
                score = "Fifteen";
            if (P1point==2)
                score = "Thirty";
            score += "-All";
        }
        if (P1point==P2point && P1point>=3)
            score = "Deuce";

        if (P1point > 0 && P2point==0)
        {
            if (P1point==1)
                P1res = "Fifteen";
            if (P1point==2)
                P1res = "Thirty";
            if (P1point==3)
                P1res = "Forty";

            P2res = "Love";
            score = P1res + "-" + P2res;
        }
        if (P2point > 0 && P1point==0)
        {
            if (P2point==1)
                P2res = "Fifteen";
            if (P2point==2)
                P2res = "Thirty";
            if (P2point==3)
                P2res = "Forty";

            P1res = "Love";
            score = P1res + "-" + P2res;
        }

        if (P1point>P2point && P1point < 4)
        {
            if (P1point==2)
                P1res="Thirty";
            if (P1point==3)
                P1res="Forty";
            if (P2point==1)
                P2res="Fifteen";
            if (P2point==2)
                P2res="Thirty";
            score = P1res + "-" + P2res;
        }
        if (P2point>P1point && P2point < 4)
        {
            if (P2point==2)
                P2res="Thirty";
            if (P2point==3)
                P2res="Forty";
            if (P1point==1)
                P1res="Fifteen";
            if (P1point==2)
                P1res="Thirty";
            score = P1res + "-" + P2res;
        }

        if (P1point > P2point && P2point >= 3)
        {
            score = "Advantage player1";
        }

        if (P2point > P1point && P1point >= 3)
        {
            score = "Advantage player2";
        }

        if (P1point>=4 && P2point>=0 && (P1point-P2point)>=2)
        {
            score = "Win for player1";
        }
        if (P2point>=4 && P1point>=0 && (P2point-P1point)>=2)
        {
            score = "Win for player2";
        }
        return score;
    }

    void SetP1Score(int number){

        for (int i = 0; i < number; i++)
        {
            P1Score();
        }

    }

    void SetP2Score(int number){

        for (int i = 0; i < number; i++)
        {
            P2Score();
        }

    }

    void P1Score(){
        P1point++;
    }

    void P2Score(){
        P2point++;
    }

    void wonPoint(String player) {
        if (player == "player1")
            P1Score();
        else
            P2Score();
    }
}